lightning/src/ln/chan_utils.rs

```diff
@@ -871,6 +890,7 @@ impl_writeable_tlv_based!(CommitmentTransaction, {
 	(8, keys, required),
 	(10, built, required),
 	(12, htlcs, vec_type),
+	(14, opt_anchors, required),
```

Comment:
Making this `required` with an even value means if you serialize out a new `CommitmentTransaction`, even if it doesn't use anchors, old versions will refuse to read it, further, we won't be able to read old `CommitmentTransaction`s. You probably want an even value (implying that old versions will refuse to read it if its present) but make it an `option` so that new versions will still read ones that don't have it. Maybe make the field an `Option<()>` to indicate if it has anchors or not. Otherwise you'll need to pull out the macro and do the read/write manually or adapt the macro to support this use-case explicitly.

Reply:
Done